Dev

JChambers13 opened the pull request:
The assessment

JChambers13 committed:
Started adding code

Assigned instances of each class

JChambers13 committed:
Update Die.cs

JChambers13 committed:
Update Die.cs

JChambers13 committed:
Update Die.cs

JChambers13 committed:
Added dieroll function

JChambers13 committed:
added code for dieRoll

JChambers13 committed:
Added code

Added code to the testing class. Also moved console.writeline to program from game. Made the dieRoll return only the sum, not each individual die.

JChambers13 committed:
Added tests to testing class

Also made minor adjustments to other classes

JChambers13 committed:
Changed Die objects to private

JChambers13 committed:
Made die rolls continuous

Added a while loop and  a bool to allow the user to roll the dice again if they choose to.

JChambers13 committed:
Updated wording of an output

JChambers13 committed:
Added encapsulation to die.cs

also temporarily commented out testing stuff

JChambers13 committed:
added comments

also made the random static to provide greater randomness

JChambers13 committed:
Removed unneccasary comment

OwenHope96 commented on CMP1903_A1_2324/Program.cs:
```diff
@@ -10,10 +12,37 @@ internal class Program
+                else //If user types anything else, dice is rolled again.
+                {
+                    Console.WriteLine("Dice will be rolled again.");
+                    //Console.ReadLine();
```
Console.ReadLine(); is commented out, left in by accident?

OwenHope96 commented on CMP1903_A1_2324/Program.cs:
```diff
@@ -10,10 +12,37 @@ internal class Program
+
+            }
+
+            Console.WriteLine("No more dice rolling. Enter anything to close the program!");
```
Unnecessary, if they already entered they do not want to roll the dice more they do not need this line of code.

OwenHope96 commented on CMP1903_A1_2324/Game.cs:
```diff
@@ -17,5 +17,30 @@ internal class Game
         //Methods
 
+        //these are three object instantiaons of die objects. They are private as they do not need to be accessed outside of the game class
+        private Die _die1 = new Die(); 
```
Instead of 3 dice could potentially roll one die 3 times by iterating through the rollDice method of the Die class.

OwenHope96 commented on CMP1903_A1_2324/Die.cs:
```diff
@@ -14,10 +14,40 @@ internal class Die
+        {
+            
+            _dieValue = rolling(rnd);
+            Console.WriteLine("Vallue of this die is: " +  dieValue);
```
Typo

OwenHope96 reviewed (approved):
All variables and methods in camel case, class names are in Pascal case though. Well commented and implemented correctly and elegantly.

Sammy226 reviewed (commented):
1) Spelling mistake on line 37 (Die) "Vallue".

2) //tRoll = testDie.rollDice();
//Debug.Assert((0 < tRoll) && (tRoll < 7), "Dice roll too big");

This is used as a comment, was this an accident. it looks like it should be code not comments.

3) You could have the program end immediately after the user types "n" just by removing: 
Console.WriteLine("No more dice rolling. Enter anything to close the program!");
Console.ReadKey();

but this is more of a personal preference thing.

4) spelling error on line 42 (Die) "b"

5) could add another data value e.g. the dice total value which adds up their value when rolled multiple times.

overall, very easy to read and understand, the comments help, its very clear and spaced out cleanly.

JChambers13 committed:
Made changes requested by reviewers

Removed unneccesary lines, added a continuous total, made the testing block actual code and not comments

JChambers13 committed:
Updated code to follow style guide

Removed unneccesary comments and made sure all opening braces are on the correct line.

Sammy226 commented:
Have the requirements been met?
Yes, the code works as intended, it works efficiently and also includes an option to re-roll the dice.

Is the code easy to read.
Yes the code is laid out clearly and concisely which makes it clear to anyone reading it, on top of that the comments make it easier for anyone who isn't completely familiar with C#.

Is the code formatted correctly according to the guidelines?
mostly, while it works fine and is easy to read, as you pointed out to me there is a small difference between the guidelines as it should be:

"code {

}"

instead of

"code
{

}"

Are there any errors?
there are a few spelling mistakes here and there which I've already pointed out. But anything that might affect the performance of the program has been dealt with and isn't a problem.

JakeH112233 reviewed (commented):
1: Have the requirements been met?
I can see that the requirements have been met and the dice is able to be rerolled without any errors or bugs.
2: Is the code formatted using the Style Guidelines correctly?
I would say it could be formatted a bit neater with the commenting and indentation used but it seems to be to the correct format of C# and functions.
3: Is the code easy to read?
Yes the code is easy to read and there are lots of comments which is helpful for viewers that may not understand C# or any code, they can still understand what is going on!
4: Are different errors handled correctly?
Errors are handled accordingly making sure that the dice values do not exceed the specified limits and also loops to ensure the game still runs even when wrong inputs are entered.

Overall, good display of OOP knowledge and code design, meeting all the requirements without any major visible errors!
